substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/heap/PhysicalMemory.java

```diff
@@ -107,6 +112,32 @@ public static UnsignedWord size() {
         return cachedSize;
     }
 
+    /**
+     * Returns the amount of used physical memory in bytes, or -1 if not supported yet.
+     */
+    public static long usedSize() {
+        if (Containers.isContainerized()) {
+            return Containers.memoryUsageInBytes();
+        }
```

Comment:
Unfortunately, `Containers.isContainerized()` will return true on most physical Linux machines the way it's implemented now. You'd have to do additional checks: e.g. `Containers.memoryLimitInBytes() > 0` to actually enter the non-containerized branches.

Reply:
Ok I see. I've adopted your suggestion and I'm now checking `Containers.memoryLimitInBytes()` instead.

Reply:
I don't think we need `Containers.memoryUsageInBytes()`. Here is what I think should work for the majority of cases: https://github.com/jerboaa/graal/commit/03b319579abcd26c0761c6f55f7811b71914f392

Reply:
ok I see what you mean. I think that's a good idea.